apps/shared/avifpng.c

```diff
@@ -17,6 +20,142 @@ typedef png_bytep png_iccp_datap;
 typedef png_charp png_iccp_datap;
 #endif
 
+// Converts a hexadecimal string which contains 2-byte character representations of hexadecimal values to raw data (bytes).
+// hexString may contain values consisting of [A-F][a-f][0-9] in pairs, e.g., 7af2..., separated by any number of newlines.
+// On success the bytes are filled and AVIF_TRUE is returned.
+// AVIF_FALSE is returned if fewer than expectedLength hexadecimal pairs are converted.
+static avifBool avifHexStringToBytes(const char * hexString, size_t hexStringLength, size_t numExpectedBytes, avifRWData * bytes)
+{
+    // Remove any trailing new line. They are not part of the numExpectedBytes.
+    while ((hexStringLength >= 1) && (hexString[0] == '\n')) {
+        ++hexString;
+        --hexStringLength;
+    }
+
+    // Preprocess the input hexString by removing a tag commonly added at encoding, if present.
+    // HEIF specification ISO-23008 section A.2.1 allows including and excluding it from AVIF files.
+    // The PNG 1.5 extension mentions the omission of this header for the modern standard eXIf chunk.
+    const char tagExif00[] = "457869660000"; // "Exif\0\0" tag encoded as a hexadecimal string.
+    const size_t tagExif00Len = 6 * 2;
+    if ((hexStringLength >= tagExif00Len) && !memcmp(hexString, tagExif00, tagExif00Len)) {
```

Comment:
I added a `(hexStringLength < (numExpectedBytes * 2))` check above and replaced the condition by `(numExpectedBytes >= (tagExif00Len / 2))` instead.

I also added a `(numExpectedBytes == 0)` check to discard empty payloads as invalid. It could be part of the condition above because with the current code it can happen only there, but it looks safer and simpler not to nest them.

Reply:
Your solution also works. `hexStringLength >= (numExpectedBytes * 2)` and `numExpectedBytes >= (tagExif00Len / 2)` implies `hexStringLength >= tagExif00Len`, so `hexStringLength -= tagExif00Len;` can be executed safely.